src/crc32c_read_le.h

```diff
@@ -8,48 +8,32 @@
 #include <cstdint>
 #include <cstring>
 
-#include "crc32c/crc32c_config.h"
-
 namespace crc32c {
 
 // Reads a little-endian 16-bit integer from bytes, not necessarily aligned.
 inline uint16_t ReadUint16LE(const uint8_t* buffer) {
-#if BYTE_ORDER_BIG_ENDIAN
-  return ((uint16_t{buffer[0]}) | (uint16_t{buffer[1]} << 8));
-#else   // !BYTE_ORDER_BIG_ENDIAN
-  uint16_t result;
-  // This should be optimized to a single instruction.
-  std::memcpy(&result, buffer, sizeof(result));
-  return result;
-#endif  // BYTE_ORDER_BIG_ENDIAN
+  return ((static_cast<uint16_t>(static_cast<uint8_t>(buffer[0]))) |
```

Comment:
`buffer` is of type `const uint8_t*`.
`buffer[0]` is of type `uint8_t`.
Hence, the explicit conversion `static_cast<uint8_t>(buffer[0])` is redundant.